subgraph/tests/matchstick/access-control/TokenRoleChanged.test.ts

```diff
@@ -0,0 +1,143 @@
+    );
+
+    tokenRoleChangedList.push(
+      createNewTokenRoleChanged(2, BigInt.fromI32(0), 0, TOKEN_OWNER_ONE, false, TOKEN_OWNER_TWO) // Token Owner Two revokes the controller access of Token Owner One to tokenId 0
+    );
+
+
+    handleTokenRoleChangedList(tokenRoleChangedList);
+  });
+
+  afterAll(() => {
+    clearStore();
+  });
+
+  describe('Assertions', () => {
+    test('Check the `tokenId` field of each TokenRoleChanged event entity', () => {
+      assert.fieldEquals(
+        'TokenRoleChanged',
+        makeEventId(0),
+        'tokenId',
+        '0'
+      );
+      assert.fieldEquals(
+        'TokenRoleChanged',
+        makeEventId(1),
+        'tokenId',
+        '1'
+      );
+      assert.fieldEquals(
+        'TokenRoleChanged',
+        makeEventId(2),
+        'tokenId',
+        '0'
+      );
+    });
+    test('Check the `role` field of each TokenRoleChanged event entity', () => {
+      assert.fieldEquals(
+        'TokenRoleChanged',
+        makeEventId(0),
+        'role',
+        '0'
+      );
+      assert.fieldEquals(
+        'TokenRoleChanged',
+        makeEventId(1),
+        'role',
+        '0'
+      );
+      assert.fieldEquals(
+        'TokenRoleChanged',
+        makeEventId(2),
+        'role',
+        '0'
+      );
+    });
+
+    test('Check the `toAddress` field of each TokenRoleChanged event entity', () => {
+      assert.fieldEquals(
+        'TokenRoleChanged',
+        makeEventId(0),
+        'toAddress',
+        TOKEN_OWNER_ONE.toString()
+      );
+      assert.fieldEquals(
+        'TokenRoleChanged',
+        makeEventId(1),
+        'toAddress',
+        TOKEN_OWNER_TWO.toString()
+      );
+      assert.fieldEquals(
+        'TokenRoleChanged',
+        makeEventId(2),
+        'toAddress',
+        TOKEN_OWNER_ONE.toString()
+      );
+    });
+
+    test('Check the `byAddress` field of each TokenRoleChanged event entity', () => {
+      assert.fieldEquals(
+        'TokenRoleChanged',
+        makeEventId(0),
+        'byAddress',
+        TOKEN_OWNER_TWO.toString()
+      );
+      assert.fieldEquals(
+        'TokenRoleChanged',
+        makeEventId(1),
+        'byAddress',
+        TOKEN_OWNER_ONE.toString()
+      );
+      assert.fieldEquals(
+        'TokenRoleChanged',
+        makeEventId(2),
+        'byAddress',
+        TOKEN_OWNER_TWO.toString()
+      );
+    });
+
+    test('Check the `status` field of each TokenRoleChanged event entity', () => {
+      assert.fieldEquals(
+        'TokenRoleChanged',
+        makeEventId(0),
+        'status',
+        'true'
+      );
+      assert.fieldEquals(
+        'TokenRoleChanged',
+        makeEventId(1),
+        'status',
+        'true'
+      );
+      assert.fieldEquals(
+        'TokenRoleChanged',
+        makeEventId(2),
+        'status',
+        'false'
+      );
+    });
```

Comment:
Similar to my previous comment, maybe here we can assert if the current token roles list in the entities is correct?

Reply:
From our discussion on Discord:

To be discussed after #168